scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5240 @@
+#  Make ShiftedScaledDistribution more efficient - only process underlying
+#   distribution parameters as necessary.
+#  Reconsider `all_inclusive`
+#  Should process_parameters update kwargs rather than returning? Should we
+#   update parameters rather than setting to what process_parameters returns?
+
+# Questions:
+# 1.  I override `__getattr__` so that distribution parameters can be read as
+#     attributes. We don't want uses to try to change them.
+#     - To prevent replacements (dist.a = b), I could override `__setattr__`.
+#     - To prevent in-place modifications, `__getattr__` could return a copy,
+#       or it could set the WRITEABLE flag of the array to false.
+#     Which should I do?
+# 2.  `cache_policy` is supported in several methods where I imagine it being
+#     useful, but it needs to be tested. Before doing that:
+#     - What should the default value be?
+#     - What should the other values be?
+#     Or should we just eliminate this policy?
+# 3.  `validation_policy` is supported in a few places, but it should be checked for
+#     consistency. I have the same questions as for `cache_policy`.
+# 4.  `tol` is currently notional. I think there needs to be way to set
+#     separate `atol` and `rtol`. Some ways I imagine it being used:
+#     - Values can be passed to iterative functions (quadrature, root-finder).
+#     - To control which "method" of a distribution function is used. For
+#       example, if `atol` is set to `1e-12`, it may be acceptable to compute
+#       the complementary CDF as 1 - CDF even when CDF is nearly 1; otherwise,
+#       a (potentially more time-consuming) method would need to be used.
+#     I'm looking for unified suggestions for the interface, not ad hoc ideas
+#     for using tolerances. Suppose the user wants to have more control over
+#     the tolerances used for each method - how do they specify it? It would
+#     probably be easiest for the user if they could pass tolerances into each
+#     method, but it's easiest for us if they can only set it as a property of
+#     the class. Perhaps a dictionary of tolerance settings?
+# 5.  I also envision that accuracy estimates should be reported to the user
+#     somehow. I think my preference would be to return a subclass of an array
+#     with an `error` attribute - yes, really. But this is unlikely to be
+#     popular, so what are other ideas? Again, we need a unified vision here,
+#     not just pointing out difficulties (not all errors are known or easy
+#     to estimate, what to do when errors could compound, etc.).
+# 6.  The term "method" is used to refer to public instance functions,
+#     private instance functions, the "method" string argument, and the means
+#     of calculating the desired quantity (represented by the string argument).
+#     For the sake of disambiguation, shall I rename the "method" string to
+#     "strategy" and refer to the means of calculating the quantity as the
+#     "strategy"?
```

Comment:
That could work. You could also prefix both, e.g. "instance method" / "fitting_method" / "fishing_method" etc.
Or instead of "strategy", you could call it "kind", like `numpy.sort` does.